src/test_report.py

```diff
@@ -96,32 +97,89 @@ def send_mail(self, email_msg, email_server):
         else:
             self._logger.log_message(logging.INFO, "Email Sent.")
 
+    def get_test_analysis(self, nodes):
+        total_runs = len(nodes)
+        total_failures = sum(node['status'] == "fail" for node in nodes)
+        total_timeouts = sum(node['status'] == "timeout" for node in nodes)
+        return total_runs, total_failures, total_timeouts
+
+    def create_email_summary(self, root):
+        def squash(item, max_length):
+            if len(item) > max_length:
+                n = (max_length - 3) / 2
+                m = max_length - 3 - n
+                item = item[:m] + '...' + item[-n:]
+            return item
+
+        columns = [
+            "test",
+            "result",
+            "build",
+            "kind"
+        ]
+        rows = [
+            (
+                revision['name'],
+                revision['status'],
+                revision['parent'],
+                revision['kind']
+            ) for revision in root
+        ]
+
+        squashed = [tuple(squash(item, 28) for item in row)for row in rows]
+
+        widths = [len(col) for col in columns]
+        for row in squashed:
+            row_widths = [len(col) for col in row]
+            widths = [max(col) for col in zip(widths, row_widths)]
+        fmt = " | ".join("{{:{w}}}".format(w=w) for w in widths)
+        sep = '-+-'.join('-' * w for w in widths)
+        headers = '\n'.join([fmt.format(*tuple(columns)), sep])
+        summaries = [fmt.format(*row) for row in squashed]
+        return headers, summaries
+
+    def create_test_report(self, root_node, child_nodes):
+        total_runs, total_failures, total_timeouts = \
+                    self.get_test_analysis(child_nodes)
+
+        template_env = jinja2.Environment(
+                            loader=jinja2.FileSystemLoader("./config/reports/")
+                        )
+        template = template_env.get_template("test-report.jinja2")
+        subject_str = "{}/{} {}: {} runs {} fails ({})".format(
+            root_node['revision']['tree'],
+            root_node['revision']['branch'],
+            root_node['revision']['describe'],
+            total_runs,
+            total_failures,
```

Comment:
```suggestion
            total_failures,
            total_timeouts,
```

Reply:
I don't think we should be listing tests that timed out in the email report, we don't have that in the current ones. It's more of an implementation detail of KernelCI.

The most important thing to show would be the number of regressions, but until we have that we should be showing the number of failures. Maybe the total number is useful too so we can have an idea of how well tests were run.  It's really important to put as little information as possible in the email reports.